source/gosling/crates/tor-interface/src/legacy_tor_client.rs

```diff
@@ -233,8 +232,8 @@ impl LegacyTorClient {
                 (
                     None,
                     controller,
-                    tor_control_passwd.clone(),
-                    Some(tor_socks_addr.clone()),
+                    std::mem::replace(tor_control_passwd, String::new()),
```

Comment:
std::mem::take would be more succinct

Reply:
but really the right way to do this would be to use [zeroize](https://crates.io/crates/zeroize)